editoast/src/modelsv2/projects.rs

```diff
@@ -0,0 +1,278 @@
+use crate::error::Result;
+use crate::models::TextArray;
+use crate::modelsv2::{DeleteStatic, Retrieve, Save};
+use crate::views::pagination::{Paginate, PaginatedResponse};
+use crate::views::projects::ProjectError;
+use crate::DbPool;
+use actix_web::web::Data;
+use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
+use diesel::sql_types::BigInt;
+use diesel::{sql_query, ExpressionMethods, QueryDsl};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::models::List;
+
+use super::Document;
+
+crate::schemas! {
+    Ordering,
+    ProjectWithStudies,
+    Project,
+    Tags,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, ModelV2, ToSchema, PartialEq)]
+#[model(table = crate::tables::project)]
+#[model(changeset(public))]
```

Comment:
```suggestion
```

You now have `flat_<field>` functions in the changeset that accept an `Option<T>` so you don't need public fields anymore.